snare/tests/test_process_links.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+import asyncio
+import sys
+from snare.cloner import Cloner
+
+
+class TestProcessLinks(unittest.TestCase):
+    def setUp(self):
+        self.root = 'http://example.com'
+        self.level = 0
+        self.max_depth = sys.maxsize
+        self.loop = asyncio.new_event_loop()
+        self.css_validate = 'false'
+        self.handler = Cloner(self.root, self.max_depth, self.css_validate)
+        self.expected_content = None
+        self.return_content = None
+
```

Comment:
Also, check the status of `new_urls` in these test cases.

Reply:
Also this case https://github.com/mushorg/snare/blob/master/snare/cloner.py#L47

Reply:
`yarl` includes all the unicode ranges of the characters in different languages as well. So, this case is highly unlikely. However, there is possibility of `UnicodeEncodeError` that we should check instead of this.

Reply:
Maybe you can mock yarl.URL and raise this exception?

Reply:
I tried to patch it and mock but no luck :( 
Any hints?

Reply:
Not sure, just try to test it.

Reply:
Did you test it?

Reply:
Done